lamindb_setup/dev/_settings_storage.py

```diff
@@ -50,25 +53,75 @@ def get_storage_region(storage_root: Union[str, Path, UPath]) -> Optional[str]:
     return storage_region
 
 
-def process_storage_arg(storage: Union[str, Path, UPath]) -> "StorageSettings":
-    storage = str(storage)  # ensure we have a string
+def create_path_with_credentials(path: UPath) -> UPath:
+    if not isinstance(path, S3Path):
+        return path
+    path = UPath(path, cache_regions=True)
+    # the below is problematic, because it will assume that all subsequent
+    # requests will then be treated as anon, and we'll never try something else
+    # if isinstance(path, S3Path) and not upath.AWS_CREDENTIALS_PRESENT:
+    #     # if we already know we don't have AWS credentials
+    #     # let's make an anon request from the get-go
+    #     return UPath(path, anon=True)
+    try:
+        path.fs.call_s3("head_bucket", Bucket=path._url.netloc)
+        upath.AWS_CREDENTIALS_PRESENT = True  # type: ignore
+    except NoCredentialsError:
+        # we don't have any credentials at all
```

Comment:
Are you sure that this is needed? If there are no credentials but they can be received from the hub, then why not use them? Now it will only return anon.

Reply:
Also if AWS_CREDENTIALS_PRESENT logic is here and not in the original create_path, then relevant parts will break further in lamindb in the case of no credentials, because it uses create_path and assumes that it treats this case.

Reply:
Thanks for pointing this out, Sergei! I will take care of this!